tools/src/format.rs

```diff
@@ -81,13 +81,21 @@ pub fn format_code(jumps: &HashMap<String, Jump>, format: FormatAs,
 
         let column = (token.start - line_start) as u32;
 
-        // This should never happen, but sometimes the analysis
-        // has bugs in it. This works around them.
```

Comment:
Hmm, should this still not happen? If so it'd be good to know why. In any case this looks good to me.

Reply:
Honestly I don't know why that comment is there. It was added by billm 3 years ago but I think that scenario does legitimately happen and that code is needed to advance `cur_datum` properly.